tests/data_context/cloud_data_context/test_datasource_crud.py

```diff
@@ -134,3 +136,110 @@ def json(self):
         # Make sure the name is populated correctly into the created datasource
         assert retrieved_datasource.name == datasource_name
         assert retrieved_datasource.config["name"] == datasource_name
+
+
+@pytest.mark.cloud
+@pytest.mark.integration
+@pytest.mark.parametrize(
+    "config_includes_name_setting",
+    [
+        pytest.param("name_supplied_separately", id="name supplied separately"),
+        pytest.param("config_includes_name", id="config includes name"),
+        pytest.param(
+            "name_supplied_separately_and_included_in_config",
+            id="name supplied separately and config includes name",
+            marks=pytest.mark.xfail(strict=True, raises=TypeError),
```

Comment:
so this currently fails?

Reply:
Yep! TypeError because you would effectively be passing the `name` param twice.